src/cryptography/hazmat/backends/openssl/x509.py

```diff
@@ -166,6 +166,58 @@ def public_bytes(self, encoding: serialization.Encoding) -> bytes:
         self._backend.openssl_assert(res == 1)
         return self._backend._read_mem_bio(bio)
 
+    def _has_signature_of(self, signer_candidate: x509.Certificate) -> bool:
+        try:
+            pubkey = signer_candidate.public_key()
+        except ValueError as e:
+            # OpenSSL is unable to get the public key
+            raise UnsupportedAlgorithm(
+                str(e), _Reasons.UNSUPPORTED_PUBLIC_KEY_ALGORITHM
+            )
+
+        signature = self.signature
+        data = self.tbs_certificate_bytes
+        if isinstance(pubkey, asymmetric.rsa.RSAPublicKeyWithSerialization):
+            pubkey.verify(
+                signature,
+                data,
+                padding=asymmetric.padding.PKCS1v15(),
+                algorithm=self.signature_hash_algorithm,  # type: ignore
```

Comment:
Why do these need ignores?

Reply:
@reaperhulk `Certificate.signature_hash_algorithm` [is of type](https://github.com/pyca/cryptography/blob/3.4.7/src/cryptography/hazmat/backends/openssl/x509.py#L115) `Optional[hashes.HashAlgorithm]` , while `pubkey.verify` [expects](https://github.com/pyca/cryptography/blob/3.4.7/src/cryptography/hazmat/primitives/asymmetric/dsa.py#L139) its `algorithm` argument to be `Union[asym_utils.Prehashed, hashes.HashAlgorithm]`.

If the signer's public key is, say, RSA, then signature with it has to have some not-`None` hash algorithm, so we are safe to pass it to `verify()`. I don't see a way how to express this better in the code.

Reply:
If you assert on its type then mypy will unwrap the optional and this will work without type ignores